### DISCUSSION TITLE
Add indoor support with level switcher

### DISCUSSION THREAD

**Discussion by Unkn0wnCat** [organicmaps/organicmaps#1421] [Ideas]
Hi all,

I think if the user is zoomed in onto a building, a selector for floors should appear on the bottom right, so the user can look at the data for individual levels. This would allow for way better indoor support and allow for users to explore the inside of buildings easily.

This is one of the big things Google Maps currently has, which none of the OSM apps have (as far as I've seen)

Let me know what you think!

**Comment by biodranik** [comment]
This is a good idea, but it requires some effort to implement in our graphics engine and in the data processing layer. Are there many mapped indoor locations in OSM already?

**Reply by Unkn0wnCat** [reply]
There are [a lot of examples already](https://wiki.openstreetmap.org/wiki/Simple_Indoor_Tagging#Tagging_examples_.28alphabetical.29) and I think if there would be a huge influx of tagged buildings if a major and well liked app like Organic Maps implements it nicely.

A few friends and I are currently getting into indoor mapping for buildings around us, and it would be a huge motivation if there would finally be a mobile app supporting it! 🙂

**Reply by Outlet2048** [reply]
I'd love to see that and it would make OM stand out when being compared to other OSM based apps

Would absolute love to see this implemented and it would motivate me to actually start indoor mapping as well.

**Reply by zorglubu** [reply]
> Are there many mapped indoor locations in OSM already?

There are +300k indoor keys
https://taginfo.openstreetmap.org/keys/?key=indoor

And to get a picture : https://indoorequal.org/#map=4.18/47.55/2.74
Yes, we would have more contributions with a mobile app showing them. It would be great in Organic Maps! ;-)

**Reply by Outlet2048** [reply]
Looks like streetcomplete will soon ask users for the level of a shop
Meaning indoor maps would soon be a lot more useful once the data has been collected
https://github.com/streetcomplete/StreetComplete/pull/3509

**Reply by biodranik** [reply]
Cool! Any contributions to OM are welcomed!

**Reply by Outlet2048** [reply]
https://www.openstreetmap.org/#map=19/53.59175/9.87434

Heres another indoor mapped area if you want another example

**Comment by pastk** [comment]
This idea could be tracked in #1308

**Comment by TheAdventurer64** [comment]
https://github.com/organicmaps/organicmaps/issues/1308#issuecomment-3073440886
